tools/single_use/24.0_permission_changes.php

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env php
+<?php declare(strict_types=1);
+/**
+ * This script should be run during the upgrade process from LORIS versions
+ * 23.x and 24.0.
+ *
+ * LORIS 24.0 introduced 3 new permissions:
+ *      - survey_accounts_view
+ *      - imaging_quality_control_view
+ *      - behavioural_quality_control_view
+ *
+ * These permissions are used in place of less precise permission codes that
+ * were used to grant access to LORIS modules.
+ * Applying this patch will allow users who had access to specific modules using
+ * the old permission codes to maintian their access after the upgrade process.
+ */
+require_once __DIR__ . "/../../vendor/autoload.php";
+require_once '../generic_includes.php';
+
+$userQuery = "SELECT u.UserID 
+FROM users u 
+JOIN user_perm_rel upr 
+ON u.ID = upr.UserID 
+JOIN permissions p 
+ON upr.permID = p.permID 
+WHERE p.code=':code'";
+
+$newPermissions = [
+    'survey_accounts_view',
+    'imaging_quality_control_view',
+    'behavioural_quality_control_view'
+];
+
+foreach ($newPermissions as $permission) {
+    $users         = [];
+    $oldPermission = '';
+    switch ($permission) {
+    case 'survey_accounts_view':
+        $oldPermission = 'user_accounts';
+        break;
+    case 'imaging_quality_control_view':
+    case 'behavioural_quality_control_view':
+        $oldPermission = 'quality_control';
+        break;
+    }
+    $users = $DB->pselectCol($userQuery, ['code' => $oldPermission]);
```

Comment:
This query is returning empty though it works when I run it in the MySQL console. @driusan @ridz1208 (or anyone else) do you have any idea why this wouldn't work? I can't figure it out.

Reply:
probably because of the quotation marks around `:code` in the query, but regardless wouldn't it be easier to do all this with an SQL statement similar to `23.0/2020-04-27-AddViolatedScansOwnSitePermission.sql` or `23.0/2018-07-23-battery_manager_permissions.sql`?

Reply:
I haven't tested this and it's not a very well written query for code, but something like it should be sufficient for an upgrade patch:

```
INSERT INTO user_perm_rel 
    SELECT userID, (SELECT permID FROM permissions WHERE code='survey_accounts_view') 
        FROM user_perm_rel WHERE permID=
            (SELECT permID FROM JOIN permissions WHERE code='user_accounts');
```

Reply:
> because of the quotation marks

Thanks, removing them worked

> wouldn't it be easier to do all this with an SQL statement

SQL isn't my strong suit and I'm short on time to get everything worked out across my remaining PRs + other responsibilities.

I agree SQL would be nicer in terms of a release process. Unfortunately your query above doesn't run. 

Given my time constraints I don't think I'm going to be able to debug this. 

Our approaches are:

1) Merge this; add a ticket to refactor into MySQL
2) Someone else can take over this PR and modify it to be SQL instead of this script
3) Merge as-is. The script is sufficient to get everything working though has the flaw of being an extra step in the release

Up to you

Reply:
I had a typo in the select statement.. ("`FROM JOIN`"). It works for me when I fix it:

```
mysql> INSERT INTO user_perm_rel SELECT userID, (SELECT permID FROM permissions WHERE code='survey_accounts_view') FROM user_perm_rel WHERE permID=(SELECT permID FROM permissions WHERE code='user_accounts');
Query OK, 42 rows affected (0.00 sec)
Records: 42  Duplicates: 0  Warnings: 0
```

Reply:
thanks that works